src/coreComponents/physicsSolvers/SolverBase.cpp

```diff
@@ -604,6 +586,51 @@ bool SolverBase::lineSearchWithParabolicInterpolation( real64 const & time_n,
   return lineSearchSuccess;
 }
 
+real64 SolverBase::assembleResidualOnlyAndCalculateNorm( real64 const & time_n,
+                                                         real64 const & dt,
+                                                         DomainPartition & domain,
+                                                         DofManager const & dofManager,
+                                                         CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                                         ParallelVector & rhs,
+                                                         string const logMessage )
+{
+  localMatrix.zero();
+  rhs.zero();
+
+  arrayView1d< real64 > const localRhs = rhs.open();
+
+  // The current implementation of this function is incomplete
+  // Ultimately, we want to modify the two calls below to only recompute the residual, and not the Jacobian matrix
+  // At that point, the localMatrix argument of this function will be removed
+
+  assembleSystem( time_n,
+                  dt,
+                  domain,
+                  dofManager,
+                  localMatrix,
+                  localRhs );
+
+  applyBoundaryConditions( time_n,
+                           dt,
+                           domain,
+                           dofManager,
+                           localMatrix,
+                           localRhs );
+
+  rhs.close();
+
+  if( getLogLevel() >= 1 && logger::internal::rank==0 && !logMessage.empty() )
+  {
+    std::cout << logMessage;
+  }
```

Comment:
This looks out of place, though I think I understand why it was done... is this just to get Line Search and residual printout closer together in the output (uninterrupted by logs from assembly functions)? Would it be too bad if we just printed this message in the Line search code before calling `assembleResidualOnlyAndCalculateNorm()`? 

On a related issue, IMO `calculateResidualNorm()` should not be printing anything, all the output in the nonlinear loop should be done by said loop directly. This would require us to return multiple values from the call (all the sub-norms to make them available for printing) so it would take a separate refactoring PR. Just noting.

Reply:
Yes, this logging was there to preserve the current output of the line search and not change our logging format (now reverted back to the original version).

I agree with you about the logging. @MelReyCG expressed interest in working on refactoring our logging strategy, maybe that will be a good opportunity to implement what you are suggesting